intro.js

```diff
@@ -150,34 +151,38 @@
           currentItem.disableInteraction = this._options.disableInteraction;
         }
 
-        if (currentItem.element != null) {
+        if (currentItem.element !== null) {
           introItems.push(currentItem);
         }
       }
 
     } else {
       //use steps from data-* annotations
       var allIntroSteps = targetElm.querySelectorAll('*[data-intro]');
+      var elmsLength = allIntroSteps.length;
+      var currentElement;
+      var disableInteraction;
+
       //if there's no element to intro
-      if (allIntroSteps.length < 1) {
+      if (elmsLength < 1) {
         return false;
       }
 
       //first add intro items with data-step
-      for (var i = 0, elmsLength = allIntroSteps.length; i < elmsLength; i++) {
-        var currentElement = allIntroSteps[i];
+      for (i = 0; i < elmsLength; i++) {
```

Comment:
And I think `i` should only be visible to this for-loop block. If we remove the previous `var i` and define `i` here.

Reply:
I think you've come across the issue exactly: `var` would not be only visible to that for-loop block; that's why jshint complains that `i` is being redefined.  We could move to ES6 and do `let`, which might not be a great idea yet, or, we could wrap these for loops in a closure, maybe even have a helper function for iterating arrays; that way `var` would be only visible in that block.  Let me know!

Reply:
@afshinm Here's what I was waiting on.  Here are some suggested ways to address this issue:

1. wrap the loops in closures:
```javascript
(function () {
  for (var i = 0; len = steps.length; i < len; i++) { 
    // etc
  }
})();
```
2. helper function to deal with loops:
```javascript
_forEach(steps, forEachFnc, endLoopFnc);
```

And I don't yet recommend introducing the ES6 variant `let`, but at some point in the future it might be a good idea.

Reply:
@bozdoz ah I see. Let's do that _forEach thing, makes more sense to me. +1 for `let` but in the future.